lib/dal/src/audit_logging.rs

```diff
@@ -184,50 +198,93 @@ pub async fn list(ctx: &DalContext) -> Result<Vec<FrontendAuditLog>> {
+    match audit_log {
+        AuditLog::V3(inner) => {
+            // TODO(nick): cache change sets.
+            let (change_set_id, change_set_name) = match inner.change_set_id {
+                Some(change_set_id) => {
+                    let change_set = ChangeSet::find(ctx, change_set_id.into())
+                        .await
+                        .map_err(Box::new)?
+                        .ok_or(AuditLoggingError::ChangeSetNotFound(change_set_id.into()))?;
+                    match change_set.status {
+                        ChangeSetStatus::Abandoned
+                        | ChangeSetStatus::Failed
+                        | ChangeSetStatus::Rejected => {
+                            trace!(?change_set.status, ?change_set.id, "skipping change set for audit log assembly due to status");
+                            return Ok(None);
+                        }
+                        ChangeSetStatus::Applied
+                        | ChangeSetStatus::Approved
+                        | ChangeSetStatus::NeedsAbandonApproval
+                        | ChangeSetStatus::NeedsApproval
+                        | ChangeSetStatus::Open => {
+                            (Some(change_set_id), Some(change_set.name.to_owned()))
+                        }
+                    }
+                }
+                None => (None, None),
+            };
+
+            // TODO(nick): cache users.
+            let (user_id, user_email, user_name) = match inner.actor {
+                Actor::System => (None, None, None),
+                Actor::User(user_id) => {
+                    let user = User::get_by_pk(ctx, user_id.into())
+                        .await
+                        .map_err(Box::new)?
+                        .ok_or(AuditLoggingError::UserNotFound(user_id.into()))?;
+                    (
+                        Some(user_id),
+                        Some(user.email().to_owned()),
+                        Some(user.name().to_owned()),
+                    )
+                }
+            };
+
+            let kind = inner.kind.to_string();
+            let deserialized_metadata = FrontendAuditLogDeserializedMetadata::from(inner.kind);
+            let (display_name, entity_type) = deserialized_metadata.display_name_and_entity_type();
+
+            Ok(Some(FrontendAuditLog {
+                display_name: display_name.to_owned(),
+                user_id,
+                user_email,
+                user_name,
+                kind,
+                entity_name: inner.entity_name,
+                entity_type: entity_type.to_owned(),
+                timestamp: inner.timestamp,
+                change_set_id,
+                change_set_name,
+                metadata: serde_json::to_value(deserialized_metadata)?,
+            }))
+        }
+        AuditLog::V2(_) | AuditLog::V1(_) => {
+            debug!("skipping older audit logs in beta...");
+            Ok(None)
+        }
```

Comment:
Just a reminder that for right now while the beta streams are active, you'll need to make new logs. Worrying about the conversations for logs that we will be throwing out later is likely wasted time. We'll see if we want to start converting if we want to see how logs live long term.